spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelBindingService.java

```diff
@@ -71,12 +75,17 @@ public ChannelBindingService(ChannelBindingServiceProperties channelBindingServi
 		List<Binding<MessageChannel>> bindings = new ArrayList<>();
 		Binder<MessageChannel, ConsumerProperties, ?> binder =
 				(Binder<MessageChannel, ConsumerProperties, ?>) getBinderForChannel(inputChannelName);
-		Class<? extends ConsumerProperties> propertiesClass = resolveConsumerPropertiesType(binder);
 		ConsumerProperties consumerProperties =
-				this.channelBindingServiceProperties.getConsumerProperties(inputChannelName, propertiesClass);
+				this.channelBindingServiceProperties.getConsumerProperties(inputChannelName);
+		if (binder instanceof ExtendedPropertiesBinder) {
+			ExtendedPropertiesBinder extendedPropertiesBinder = (ExtendedPropertiesBinder) binder;
+			Object extension = extendedPropertiesBinder.getExtendedConsumerProperties(inputChannelName);
+			ExtendedConsumerProperties extendedConsumerProperties = new ExtendedConsumerProperties(extension);
+			BeanUtils.copyProperties(consumerProperties, extendedConsumerProperties);
+			consumerProperties = extendedConsumerProperties;
```

Comment:
But, when `setConsumer()` is called, and it contains an entry for this channel, __that__ instance is then returned (from the map).

But, as I said, I don't see `setConsumer()` being called anyplace so maybe it's moot. However, if if that will be possible in the future, the props should really be copied to a new instance too.